hera_cal/filter_parser.py

```diff
@@ -0,0 +1,132 @@
+from copy import deepcopy
```

Comment:
seems like this could all go under `hera_cal.utils` or `hera_cal.io` rather than being its own submodule. Lmk what you think: I generally tend to have a high activation energy for a new module.

Reply:
It seems that most of the time, the argparsers stick with their code: abscal for example. In this case, we have two different pieces of code that do xtalk filtering or delay filtering and they both inherit vis_clean.

Maybe the right thing to do here is to put the internal argparsers in vis_clean and then we put the exposed argparsers in xtalk and delay filtering. What do you think?

Reply:
that sounds reasonable